content/security.md

```diff
@@ -0,0 +1,380 @@
+2. This method doesn't modify the other user, but uses it as a target; for example, it could be a method for sending a private message, or adding a user as a friend.
+
+#### Make methods as specific as possible
+
+The best way to make your app secure is to understand all of the possible inputs that could come from an untrusted source, and make sure that they are all handled correctly. The easiest way to understand what inputs could come from the client is to restrict them to as small of a space as possible. This means your methods should all be specific actions, and shouldn't take a multitude of options that change the behavior in significant ways. The end goal is that you can easily look at each method in your app and validate or test that it is secure. Here's a secure example method from the Todos example app:
+
+```js
+Lists.methods.makePrivate = new Method({
+  name: 'Lists.methods.makePrivate',
+  schema: new SimpleSchema({
+    listId: { type: String }
+  }),
+  run({ listId }) {
+    if (!this.userId) {
+      throw new Meteor.Error('Lists.methods.makePrivate.notLoggedIn',
+        'Must be logged in to make private lists.');
+    }
+
+    const list = Lists.findOne(listId);
+
+    if (list.isLastPublicList()) {
+      throw new Meteor.Error('Lists.methods.makePrivate.lastPublicList',
+        'Cannot make the last public list private.');
+    }
+
+    Lists.update(listId, {
+      $set: { userId: this.userId }
+    });
+
+    Lists.userIdDenormalizer.set(listId, this.userId);
+  }
+});
+```
+
+You can see that this method does a _very spefific thing_ - it just makes a single list private. An alternative would have been to have a method called `setPrivacy`, which could set the list to private or public, but it turns out that in this particular app the security considerations for the two related operations - `makePrivate` and `makePublic` - are very different. By splitting our operations into different methods, we make each one much clearer. It's obvious from the above method definition which arguments we accept, what security checks we perform, and what operations we do on the database.
+
+However, this doesn't mean you can't have any flexibility in your methods. Let's look at an example:
+
+```js
+const Meteor.users.methods.setUserData = new Method({
+  name: 'Meteor.users.methods.setUserData',
+  schema: new SimpleSchema({
+    fullName: { type: String, optional: true },
+    dateOfBirth: { type: Date, optional: true },
+  }),
+  run(fieldsToSet) {
+    Meteor.users.update(this.userId, {
+      $set: fieldsToSet
+    });
+  }
+});
+```
+
+The above method is great because you can have the flexibility of having some optional fields and only passing the ones you want to change. In particular, what makes it possible for this method is that the security considerations of setting one's full name and date of birth are the same - we don't have to do different security checks for different fields being set.
```

Comment:
I would probably re-emphasize that it's best to not pass `$set` in as an argument to the method, but instead to make various methods for various operations.